superset/daos/tag.py

```diff
@@ -257,3 +266,90 @@ def get_tagged_objects_for_tags(
                 for obj in saved_queries
             )
         return results
+
+    @staticmethod
+    def user_favorite_tag(tag_id: int) -> None:
+        """
+        Marks a specific tag as a favorite for the current user.
+        This function will find the tag by the provided id,
+        create a new UserFavoriteTag object that represents
+        the user's preference, add that object to the database
+        session, and commit the session. It uses the currently
+        authenticated user from the global 'g' object.
+        Args:
+            tag_id: The id of the tag that is to be marked as
+                    favorite.
+        Raises:
+            Any exceptions raised by the find_by_id function,
+            the UserFavoriteTag constructor, or the database session's
+            add and commit methods will propagate up to the caller.
+        Returns:
+            None.
+        """
+        tag = TagDAO.find_by_id(tag_id)
```

Comment:
we should check tag is None here and throw.
Then in the controller you catch the TagNotFoundException and return 404 http code